### ISSUE TITLE
How can I add dynamic height ?

### ISSUE THREAD

**Issue by muhammadwafa** [Flipkart/recyclerlistview#465]
I have a simple chat application and the chat messages **are different** one message can be o**ne line** the other message can **be 10 line or 9 line or 5 or 20 line**, so How to overcome this situation with this nice package ?? any help.

**Comment by muhammadwafa** [comment]
@naqvitalha 
the forceNonDeterministicRendering={true} never helped me.
I have a chat app, there is dynamic height for any messages item.

**Comment by naqvitalha** [comment]
Just give approx values. What problem are you facing? Shift animations?

**Comment by muhammadwafa** [comment]
Hi @naqvitalha thank you to reply,
Problem is there auto jumping motions and also loads data very slowly and also make the layout bad.

I have chat application one item in right the other in left, but this makes the item from right to left.

**Comment by muhammadwafa** [comment]
Could you please tell me what is the approx value ??

**Comment by muhammadwafa** [comment]
will this help `canChangeSize` ??

**Comment by muhammadwafa** [comment]
@naqvitalha, Is still faster this module than react-native with dynamic height ????

**Comment by dolphinflow86** [comment]
Facing same issue here.
@muhammadwafa  Did you solve this issue?

I have turn on forceNonDeterministicRendering={true} and it shows chat messages with calculated height correctly but slowly. 

I wish I could know the height of each chat message before we connect data with dataProvider.

**Comment by muhammadwafa** [comment]
Did you check the `canChangeSize ` ?

I left this and I am using `flatlist` I dont know that this will be faster than `flatlist` with dynamic list or not

**Comment by naqvitalha** [comment]
Non deterministic mode at this point is almost as fast as the regular one. Please make sure you've read the perf guide.

**Comment by muhammadwafa** [comment]
@naqvitalha thank you, please kindly briefly let me know that items with dynamic height can it be still faster than `flatlist` or not ????

I cant what did you mean sorry as I am not a native English speaker

**Comment by naqvitalha** [comment]
Yes, simply put it can be faster even with dynamic heights?

**Comment by muhammadwafa** [comment]
Thank you very very much so please tell which of this prop will be use for dynamic height?

1. `forceNonDeterministicRendering={true} `
2. `canChangeSize={true}`

**Comment by legion-zver** [comment]
@muhammadwafa if you create react native application - use library for calc text height

**Comment by muhammadwafa** [comment]
Hi @legion-zver, Could you please explain more ?? how can I calculate height ???

**Comment by davx1992** [comment]
@muhammadwafa  had same issue. Spent lot of time to get fast ui. In the end decided to calculate text height using https://www.npmjs.com/package/react-native-text-size and save this height in db with message. So when I am rendering data, I take this text height and use constants from your ui (paddings, margins e.t.c) to calculate total message height.

**Comment by muhammadwafa** [comment]
@davx1992 thank you, but what about the first time items gets added, and could you please share your code

**Comment by davx1992** [comment]
@muhammadwafa when I add new message I save it first and then when it is saved using saved entity to add to list.
Do not have public repository for my project but will share you some snippets.
https://gist.github.com/davx1992/8f302e96820cee1a09dde82e00e0878c

There I am calculating message height based on Text height.

**Comment by muhammadwafa** [comment]
thank you, and is that helped you as well?

**Comment by davx1992** [comment]
Yes, initially I implemented this for Flat List, as in flat list to scroll height is mandatory. However due to performance issues, decided to switch to RecyclerViewList, and now it is much faster.

**Comment by sinhpn92** [comment]
> Thank you very very much so please tell which of this prop will be use for dynamic height?
> 
> 1. `forceNonDeterministicRendering={true} `
> 2. `canChangeSize={true}`

When I have used this props, the row width and height will be recalculate both :((.

**Comment by velidan** [comment]
```
forceNonDeterministicRendering={true}
canChangeSize={true}
```

In the case of React Native Web these props cause infinity re-rendering issue so it looks like it's impossible to achieve dynamic items height

**Comment by fukemy** [comment]
[ghost](https://github.com/ghost) you CAN NOT USE dynamic height because this lib using COLLECTIONVIEW, this always need calculating the view size, the the reason [naqvitalha](https://github.com/naqvitalha) can not answer u

**Comment by imyanglu** [comment]
> @MuhammadWafa had same issue. Spent lot of time to get fast ui. In the end decided to calculate text height using https://www.npmjs.com/package/react-native-text-size and save this height in db with message. So when I am rendering data, I take this text height and use constants from your ui (paddings, margins e.t.c) to calculate total message height.


Does expo have a similar library?
